chapters/traversal.md

```diff
@@ -1,83 +1,89 @@
 # Understand Jina Recursive Document Representation
 
-In Jina, each Document is represented as a recursive representation (tree).
+This guide explains how Documents are represented within the Jina framework. It is especially relevant for understanding how segmenters, crafters and rankers work.
 
-A rooted recursive representation has a root node and every node has X children.
-In Jina, the root node is the document itself, while the *left* & *right* children are referred to as *chunks* and *matches* respectively.
+## General overview
+
+In Jina, the term Document refers to more than just a simple dataset. A Document is used for both indexing and querying. A search query is also a Document itself.
+
+When you index a Document in Jina, it can be represented as a rooted recursive representation (tree). The representation consists of a *root node* and several *child nodes*. In Jina, the root node is the original Document itself, while the *left* and *right* children are referred to as *chunks* and *matches* respectively. The figure below illustrates a basic Document structure that consists of a Document (root node) as well as two child nodes (chunks and matches).
 
 ![rooted-binary-tree](./images/overview.png)
 
-The above image illustrates a basic document structure: A document (root node) and the two possible child nodes: *chunks* and *matches*.
-chunks are a sequence of documents which is attached to the root document with a higher `granularity` degree. `matches` is a sequence of documents which are semantically related to the root document.
-We'll dive into these concepts in this chapter:
+The two terms chunks and sequence are still a bit unclear. In short, chunks are a sequence of Documents which can be attached to any Document with a higher `granularity` degree. In the figure above this is the root node. 
+
+Similarly, this applies to `matches`. `matches` is a sequence of Documents which are semantically related to the higer-level Document (the root node in the figure above). We will dive into these concepts in more detail below.
 
 - [Chunks](#chunks)
 - [Matches](#matches)
 - [Let's go deeper: Recursive Document Representation](#lets-go-deeper-recursive-document-representation)
 - [Document Traversal with traversal paths](#document-traversal-with-traversal-paths)
 
-
 ## Chunks
 
-Each Jina Document (potentially) consists of a list of chunks. A `chunk` is a small semantic unit of a Document, like a sentence or a 64x64 pixel image patch.
+### Understanding chunks
+
+Each Jina Document (potentially) consists of a list of *chunks*. A single chunk is a small semantic unit of a Document, like a sentence or a 64x64 pixel image patch.
```

Comment:
Mention that a chunk is fixed size unit of image documents and a semantic (need not be fixed size) for text documents - since length of two complete sentences or paragraphs can be different

Reply:
This information is confusing for me. Can you be so kind and explain that a bit further, @rutujasurve94 ?

Reply:
A single chunk is a small semantic unit of a Document, like a sentence or a 64x64 pixel image patch.

Here:
Let's only consider text:
A semantic unit or chunk could be a sentence. If we have a paragraph and break it into chunks (sentences), these sentences won't all be of fixed length, some are short / long.  Hence it isn't a fixed sized chunk

For images:
If we have 256x256 size image and want to break even into 64x64 size patches, we are for sure dividing into fixed size chunks

Reply:
Thank you, I have updated this part